frc_characterization/arm_characterization/data_analyzer.py

```diff
@@ -965,6 +1000,22 @@ def _calcGains(kv, ka, qp, qv, effort, period):
     R = np.diag(1.0 / np.square(r))
     K = frccnt.lqr(dsys, Q, R)
 
+    if position_delay > 0:
+        # This corrects the gain to compensate for measurement delay, which
+        # can be quite large as a result of filtering for some motor
+        # controller and sensor combinations.
+        #
+        # We use equation (11) from
+        # https://www.textroad.com/pdf/JBASR/J.%20Basic.%20Appl.%20Sci.%20Res.,%202(3)2607-2618,%202012.pdf
+        # (Note that equation (11) has a typo of P^-1 when they meant R^-1.)
```

Comment:
This isn't actually equation (11). That paper only provides a continuous version of the formula, while we're using a discrete version. I put a derivation for it in E.4.2 "Time delay compensation, discrete case" in https://file.tavsys.net/control/controls-engineering-in-frc.pdf.